channel.go

```diff
@@ -218,6 +229,12 @@ func NewChannel(serviceName string, opts *ChannelOptions) (*Channel, error) {
 		LogField{"chID", chID},
 	)
 
+	if opts.IdleCheckInterval > 0 && opts.MaxIdleTime <= 0 {
+		logger.Warn("To enable automatically removing idle connections, you must " +
```

Comment:
this should return an error, no point allowing users to create a channel with invalid config.

Suggest making an unexported function on `ChannelOptions` that does this,
```go
func (o *ChannelOpts) validateIdleCheck() error {
  ...
}

```

that way we can just return an error from validation:
```go
if err := opts.validateIdleCheck(); err != nil {
  return nil, err
}
```

Reply:
Done